core/src/main/java/org/neo4j/graphalgo/core/heavyweight/HeavyCypherGraphFactory.java

```diff
@@ -88,8 +260,48 @@ public boolean visit(Result.ResultRow row) throws RuntimeException {
                 matrix.addOutgoing(source, target, relId);
                 return true;
             }
-        });
+        }
+        RelationshipRowVisitor visitor = new RelationshipRowVisitor();
+        api.execute(setup.relationshipStatement, params(offset, batchSize)).accept(visitor);
+        return new Relationships(offset, visitor.rows, matrix, relWeigths);
+    }
+
+    private Nodes loadNodes(long offset, int batchSize) {
+        int capacity = batchSize == NO_BATCH ? INITIAL_NODE_COUNT : batchSize;
+        final IdMap idMap = new IdMap(capacity);
+
+        boolean hasNodeWeights = !setup.loadDefaultNodeWeight();
+        final WeightMapping nodeWeights = newWeightMapping(hasNodeWeights, setup.nodeDefaultWeight, capacity);
+
+        boolean hasNodeProperty = !setup.loadDefaultNodeProperty();
+        final WeightMapping nodeProps = newWeightMapping(hasNodeProperty, setup.nodeDefaultPropertyValue, capacity);
+
+        class NodeRowVisitor implements Result.ResultVisitor<RuntimeException> {
+            private long rows;
+            @Override
+            public boolean visit(Result.ResultRow row) throws RuntimeException {
+                rows++;
+                long id = row.getNumber("id").longValue();
+                idMap.add(id);
+                if (hasNodeWeights) nodeWeights.set(id, row.get("weight"));
```

Comment:
Can we have all the expected row variable names as constants somewhere? Same for `source`, `target`, `value` and `id`?